### DISCUSSION TITLE
I need to import existing data

### DISCUSSION THREAD

**Discussion by daveit** [supabase/supabase#1987] [Questions]
Hi there,
I have existing data (csv) to import, but it needs to retain the existing ID's to retain the relationships.
I can import by copying and pasting, but I can't see how to change the imported existing ID column to be a primary key ?
Am I missing something ?

Also when I choose a CSV file (very simple format - ID,Name) it fails, but if I open it in Excel and copy and paste the content it works ??  Any idea why it is failing, it seems to take the ID,Name and only sees it as 1 column!

TIA, Dave

**Comment by silentworks** [comment]
Hi @daveit, what does your database table look like that you are trying to import the data into? You can set your primary key column to not be autogenerated and you should be able to use your own IDs.

Can you provide a sample CSV (doesn't have to be the real data) and instruction as to how you are trying to upload it please?
